### PR TITLE
Polkit auth

### PR DESCRIPTION
**Pull request by Cimbali**
## Description

- Check polkit authorization for all actions that affect tunnel setup and routing:
    - activate
    - deactivate
    - add app to excluded cgroups
    - add pid to excluded cgroups
    - clear excluded cgroups
- When checking authorization, check whether DBus message _sender_ is authorized, optionally using a password popup+
   - Done by inheriting [`QDbusContext`](https://doc.qt.io/qt-6/qdbuscontext.html) on the DBus service to access info about DBus message sender
- Limit capabilities of daemon process to minimum required (CAP_NET_ADMIN and CAP_SETUID)
- Remove binary path from service file that isn’t used to start the daemon (it only serves to point to the dbus service file)

When connecting and disconnecting the VPN we now get an authorization popup as expected. If we try to change firewalled apps on the fly while connected (only via dbus messages) authorization is also checked.

## Reference

Follow-up on #7055

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [x] I have added thorough tests where needed

### CODE REVIEW

**Comment by Cimbali** [comment]
I’ve had to add e2fa20a which is an unrelated build fix.

File: qtglean/src/lib.rs [hunk 1]
```diff
@@ -96,10 +92,6 @@ pub extern "C" fn glean_test_reset_glean(is_telemetry_enabled: bool, data_path:
         // Whether Glean should schedule “metrics” pings for you
         use_core_mps: true,
         trim_data_to_registered_pings: false,
-        // Default level is "Trace"
-        log_level: None,
-        // Default is 15 pings every 60 seconds
-        rate_limit: None
```
[the remark this answers is not in the document]
Comment (brizental) [reply qtglean/src/lib.rs:102]: Huh, weird. That was merged five days ago.

[the remark this answers is not in the document]
Comment (Cimbali) [reply qtglean/src/lib.rs]: Found the issue, it was my revert commit that somehow reverted submodules too… I’ve rebased to keep as-is and drop this hacky commit.

[the remark this answers is not in the document]
Comment (brizental) [reply qtglean/src/lib.rs]: Thanks!

**Comment by oskirby** [comment]
We are currently in the process of refactoring the D-Bus API authentication in PR #7110 and we are trying to move away from using polkit. At present the direction I am hoping to go in is checking whether the client has the `CAP_NET_ADMIN` capability granted.

I am also unsure about how I feel about launching user authentication prompts in order to activate the VPN, as this would be a significant change to the user experience and would interfere with use cases like the start-on-boot feature.

**Comment by Cimbali** [comment]
> We are currently in the process of refactoring the D-Bus API authentication in PR #7110 

Fair enough, I missed that PR when writing up this one


> I am also unsure about how I feel about launching user authentication prompts in order to activate the VPN [...] 

I was hoping there would be a way down the road to have more or less strict polkit policies same as e.g. NetworkManager

**Comment by dreirund** [comment]
*[Side-Topic] for anyone interested: Context of this: https://www.openwall.com/lists/oss-security/2023/08/03/1.*

**Pull request closed without merging**